src/CodeFormatter/FormatOptions.cs

```diff
@@ -117,21 +108,24 @@ private static ImmutableArray<string> InitializeCopyrightHeaderText(string copyr
         private ImmutableDictionary<string, bool> BuildRuleMapFromOptions()
         {
             _ruleMap = ImmutableDictionary<string, bool>.Empty;
-            if (EnabledRules != null)
+
+            var propertyBag = OptionsHelper.BuildDefaultPropertyBag();
+            if (!string.IsNullOrEmpty(OptionsFilePath))
             {
-                foreach (string rule in EnabledRules)
-                {
-                    _ruleMap = _ruleMap.SetItem(rule, true);
-                }
+                propertyBag.LoadFrom(OptionsFilePath);
             }
 
-            if (DisabledRules != null)
+            propertyBag = (PropertyBag)propertyBag["CodeFormatterRules.Options"];
+            foreach (string key in propertyBag.Keys)
```

Comment:
This kind of direct access of a property bag is not typical (usually, you use a descriptor object as a key). In this case, we're munging the formatting rules code to use this mechanism. This should wither away once all code is converted to analyzers.